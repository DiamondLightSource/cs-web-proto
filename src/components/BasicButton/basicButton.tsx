import React from "react";
import { PVWidget, PVWidgetInterface } from "../Widget/widget";
import classes from "./basicButton.module.css";
import { Link } from "react-router-dom";

export interface BasicButtonProps {
  text: string;
  style?: object;
  image?: string;
  shape?: string;
}

export const BasicButtonComponent = (props: BasicButtonProps): JSX.Element => {
  if (props.image !== undefined) {
    return (
      <Link to={'/dynamic/ionpExample/{"device":"SR03A-VA-IONP-01"}'}>
        <div className={classes.image}>
          <button>
            <img src={props.image} alt={props.image}></img>
            {props.text}
          </button>
        </div>
      </Link>
    );
  } else {
    return (
      <div>
        <button>{props.text}</button>
      </div>
    );
  }
};

export interface BasicButtonWidgetProps {
  text: string;
  style?: object;
  image?: string;
}

export const BasicButtonWidget = (
  props: BasicButtonWidgetProps
): JSX.Element => {
  return (
    <BasicButtonComponent
      text={props.text}
      style={props.style}
      image={props.image}
    />
  );
};

export const BasicButton = (
  props: BasicButtonWidgetProps & PVWidgetInterface
): JSX.Element => <PVWidget baseWidget={BasicButtonWidget} {...props} />;
