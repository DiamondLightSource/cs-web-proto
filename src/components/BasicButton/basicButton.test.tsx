import React from "react";
import { BasicButtonComponent, BasicButtonProps } from "./basicButton";
import { configure, shallow, ShallowWrapper } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { create, ReactTestRenderer } from "react-test-renderer";

configure({ adapter: new Adapter() });

let snapshot: ReactTestRenderer;
let basicButtonWrapper: ShallowWrapper<BasicButtonProps>;
beforeEach((): void => {
  const basicButton = <BasicButtonComponent text={"hello"} />;
  snapshot = create(basicButton);
  basicButtonWrapper = shallow(basicButton);
});

describe("<BasicButton />", (): void => {
  test("it matches the snapshot", (): void => {
    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  test("it renders a basic element", (): void => {
    const button = basicButtonWrapper.find("button");
    expect(button.get(0).type).toEqual("button");
    expect(button.text()).toEqual("hello");
  });
});
