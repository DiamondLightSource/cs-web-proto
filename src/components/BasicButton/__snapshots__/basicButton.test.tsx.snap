// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasicButton /> it matches the snapshot 1`] = `
<div>
  <button>
    hello
  </button>
</div>
`;
